app/src/features/projects/create/CreateProjectPage.tsx

```diff
@@ -661,21 +658,36 @@ const CreateProjectPage: React.FC = () => {
     return <CircularProgress className="pageProgress" size={40} />;
   }
 
+  /**
+   * Intercepts all navigation attempts (when used with a `Prompt`).
+   *
+   * Returning true allows the navigation, returning false prevents it.
+   *
+   * @param {History.Location} location
+   * @return {*}
+   */
   const handleLocationChange = (location: History.Location, action: History.Action) => {
-    if (!openCancelDialogProps.open) {
+    if (!dialogContext.yesNoDialogProps.open) {
       // If the cancel dialog is not open: open it
-      setOpenCancelDialogProps({ ...openCancelDialogProps, onYes: () => history.push(location.pathname), open: true });
+      dialogContext.setYesNoDialog({
+        ...defaultCancelDialogProps,
+        onYes: () => {
+          dialogContext.setYesNoDialog({ open: false });
+          history.push(location.pathname);
+        },
+        open: true
+      });
       return false;
     }
 
-    // If the cancel dialog is already open and a location change action is triggered by `handleDialogYes`: allow it
+    // If the cancel dialog is already open and another location change action is triggered: allow it
+    // This should only happen
```

Comment:
is this sentence supposed to continue and say more perhaps? `this should only happen....`